torch/csrc/jit/script/exit_transforms.cpp

```diff
@@ -76,11 +78,58 @@ struct ExitTransformer {
     false_val_ = graph_->insertConstant(false);
   };
 
-  void run() {
+  void run(Transform transform_kind) {
+    if (transform_kind == Transform::Returns) {
+      current_exit_kind_ = prim::ReturnStmt;
+      convertReturnOutputsToReturnStmts(graph_->block());
+    } else {
+      current_exit_kind_ = prim::LoopContinuation;
+      convertLoopOutputsToContinuations(graph_->block());
+    }
     transformExits(graph_->block());
   }
 
  private:
+  // The Logic for the exiy transform simplifies if the block outputs
+  // are converted to Exits before running, because you do not
+  // have to handle an exit that could have maybe exited, could have not exited,
+  // or must have exited. Now, it must have to have exited.
+  static void convertBlockOutputsToNode(Block* block, Symbol kind) {
+    auto ret_node = block->return_node();
+    auto node = block->owningGraph()->create(kind, 0)->insertBefore(ret_node);
+    for (auto inp : ret_node->inputs()) {
+      node->addInput(inp);
+    }
+    removeOutputs(block);
+  }
+
+  static void convertLoopOutputsToContinuations(Block* block) {
+    for (Node* n : block->nodes()) {
+      for (Block* b : n->blocks()) {
+        convertLoopOutputsToContinuations(b);
+      }
+    }
+    if (owningNodeKind(block) == prim::Loop) {
+      convertBlockOutputsToNode(block, prim::LoopContinuation);
+    }
+  }
+
+  static bool isGraphOrClosureBlock(Block* block) {
+    return owningNodeKind(block) == prim::Function ||
```

Comment:
Nitpick: even though `owningNodeKind` correctly handles blocks without the owning node, I'd swap the expressions in this `||` statement.